### PR TITLE
Add hipMemAdvise operations

### PR DESCRIPTION
**Pull request by davidbeckingsale**

### CODE REVIEW

**Comment by MrBurmark** [comment]
yay

**Review by mcfadden8 — DISMISSED** [review]

File: src/umpire/op/HipAdviseOperation.hpp [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+namespace umpire {
+namespace op {
+
+class HipAdviseOperation : public MemoryOperation {
+ public:
+  HipAdviseOperation(hipMemoryAdvise a);
+  /*!
+   * @copybrief MemoryOperation::apply
+   *
+   * Uses cudaMemAdvise to set data as accessed by the appropriate device.
```
Comment (kab163) [inline src/umpire/op/HipAdviseOperation.hpp]: cudaMemAdvise?

Comment (davidbeckingsale) [reply src/umpire/op/HipAdviseOperation.hpp]: Copy paste lol - good catch!

**Review by kab163 — APPROVED** [review]
Looks good!

**Pull request merged**